monitor/io.py

```diff
@@ -0,0 +1,43 @@
+import os
+import subprocess
+import numpy as np
+from tempfile import mkstemp
+from shutil import move, rmtree
+
+######### ----------------------------------------###########
+
+def proc_subprocess(executing_arguments, dir):
+    proc = subprocess.Popen( ' '.join(executing_arguments),
+                            shell=True,
+                            stderr=subprocess.PIPE,
+                            stdout=subprocess.PIPE )
+    retvals = proc.communicate()
+
+    stdout = retvals[0]
+    stderr = retvals[1]
+    returncode = proc.returncode
+
+    with open(dir+'log_file', "a" ) as logfile:
+        logfile.write( stderr+stdout )
+
+######### ----------------------------------------###########
+
+def replace(file_path, pattern, subst):
```

Comment:
As @jhamman stated before, I have a different function to do this that doesn't require you to copy and delete so many files: https://github.com/lizaclark/overtheloop/blob/feature/mp_qc/tools/bin/ecflow/core/model_tools.py#L6-L23